packages/plugins/Calendar/src/SiteAdaptor/components/CountDownTimer.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import { makeStyles } from '@masknet/theme'
+import differenceInSeconds from 'date-fns/differenceInSeconds'
+
+const useStyles = makeStyles()((theme) => ({
+    timer: {
+        fontSize: '12px',
+        fontWeight: 400,
+        lineHeight: '16px',
+        background: theme.palette.maskColor.bg,
+        borderRadius: '4px',
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+        padding: '2px',
+    },
+}))
+
+interface CountDownTimerProps {
+    targetDate: Date
+}
+
+export function CountdownTimer({ targetDate }: CountDownTimerProps) {
+    const calculateRemainingTime = useCallback(
+        (targetDate: Date) => {
+            const currentDate = new Date()
+            const difference = differenceInSeconds(targetDate, currentDate)
+            return difference > 0 ? difference : 0
+        },
+        [targetDate],
+    )
+
+    const [remainingTime, setRemainingTime] = useState(calculateRemainingTime(targetDate))
```

Comment:
Avoid doing complicate calculating on every render.
```suggestion
    const [remainingTime, setRemainingTime] = useState(() => calculateRemainingTime(targetDate))
```